### PR TITLE
Implement real time updates of user

### PR DESCRIPTION
**Pull request by cp-sneha-s**
### Purpose
- Implement Real time updates of user

### Summary of Changes

- Refactor app state controller bloc
- Implement real time updates for user
- 
### Test steps

- Run the app and test it by changing space 
- 
### Conformity

- [ ] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [ ] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [ ] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [ ] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [ ] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [ ] Reused code (if the same code was written twice, made it common and reused it at both places).
- [ ] Removed unused or commented code if not required.
- [ ] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [ ] Localized user-facing strings.
- [ ] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [ ] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [ ] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [ ] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [ ] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [ ] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [ ] Marked the PR as ready before submitting it for review.

### CODE REVIEW

File: lib/data/bloc/user_state/space_user_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    await _cancelStreamSubscription();
+  }
+
+  Future<void> _cancelStreamSubscription() async {
+    await _leaveRepo.cancel();
+    await _employeeRepo.cancel();
+  }
+
+  Future<void> _resetStreamSubscription() async {
+    await _leaveRepo.reset();
```
Comment (cp-pratik-k) [inline lib/data/bloc/user_state/space_user_bloc.dart:103]: cancel subscription on bloc close function.

File: lib/data/bloc/user_state/space_user_event.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  List<Object?> get props => [];
+}
+
+class CheckUserEvent extends SpaceUserEvent {
+  final Employee? employee;
+
+  CheckUserEvent({required this.employee});
+
+  @override
+  // TODO: implement props
```
Comment (cp-pratik-k) [inline lib/data/bloc/user_state/space_user_event.dart:19]: remove comments

File: lib/main.dart [hunk 3]
```diff
@@ -44,12 +47,21 @@ class MyApp extends StatelessWidget {
...
-    return BlocProvider(
-      create: (_) =>
-          _networkConnectionBloc..add(NetworkConnectionObserveEvent()),
+    return MultiBlocProvider(
+      providers: [
+        BlocProvider(
+          create: (_) =>
+              _networkConnectionBloc..add(NetworkConnectionObserveEvent()),
+        ),
+        BlocProvider(
```
Comment (cp-pratik-k) [inline lib/main.dart:56]: why we manage here instead of dashboard

File: lib/ui/shared/appbar_drawer/drawer/bloc/app_drawer_bloc.dart [hunk 4]
```diff
@@ -46,16 +49,18 @@ class DrawerBloc extends Bloc<DrawerEvents, DrawerState> {
...
     try {
-      final spaceUser = await _employeeService.getEmployeeBySpaceId(
-          spaceId: event.space.id, userId: _userManager.userUID!);
-      if (spaceUser != null) {
-        await _userManager.setEmployeeWithSpace(
-            space: event.space, spaceUser: spaceUser);
+      _spaceManager.setCurrentSpaceId(event.space.id);
+      // final spaceUser = await _employeeService.getEmployeeBySpaceId(
+      //     spaceId: event.space.id, userId: _userManager.userUID!);
+      // if (spaceUser != null) {
```
Comment (cp-pratik-k) [inline lib/ui/shared/appbar_drawer/drawer/bloc/app_drawer_bloc.dart:55]: clean this

**Comment by cp-pratik-k** [comment]
sign out not works

**Review by cp-pratik-k — CHANGES_REQUESTED** [review]

File: lib/data/bloc/user_state/space_user_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  SpaceUserBloc(this._employeeRepo, this._leaveRepo, this._userStateNotifier,
+      this._spaceService, this._spaceManager)
+      : super(SpaceUserInitialState()) {
+    on<CheckSpaceEvent>(_checkUserStatus);
+    on<CheckUserEvent>(_updateUserData);
+    on<DeactivateUserEvent>(_deactivateUser);
+    if (_userStateNotifier.state == UserState.spaceJoined) {
+      add(CheckSpaceEvent());
+    }
+    _spaceManager.addListener(() {
```
Comment (cp-pratik-k) [inline lib/data/bloc/user_state/space_user_bloc.dart:37]: remove listner on bloc dispose
